client/components/track-clicks/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import storeShape from 'react-redux/lib/utils/storeShape';
+import { cloneElement, Children, Component, PropTypes } from 'react';
+import { constant, isFunction, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import pick from 'lib/pick';
+import { GUIDED_TOUR_TRACKED_CLICK } from 'state/action-types';
```

Comment:
For a moment I thought of playing with dynamic action types, like

```js
`TRACKED_COMPONENT_${ eventType.toUpperCase() }`
```

but I'm pretty sure that's a terrible idea. How about:

```js
{
  type: COMPONENT_INTERACTION_TRACKED,
  eventType: 'click',
  …
}
```

?